Marlin/SanityCheck.h

```diff
@@ -439,9 +439,9 @@
  */
 #if WATCH_TEMP_PERIOD > 500
   #error WATCH_TEMP_PERIOD now uses seconds instead of milliseconds.
-#elif DISABLED(THERMAL_PROTECTION_HOTENDS) && (defined(WATCH_TEMP_PERIOD) || defined(THERMAL_PROTECTION_PERIOD))
+#elif (DISABLED(THERMAL_PROTECTION_HOTENDS) || WATCH_TEMP_PERIOD <= 0) && (defined(WATCH_TEMP_PERIOD) || (defined(THERMAL_PROTECTION_PERIOD) && WATCH_TEMP_PERIOD > 0))
```

Comment:
The original reason for these tests was to deal with older configurations that didn't have `THERMAL_PROTECTION_HOTENDS` or `THERMAL_PROTECTION_BED` options, so the other options could be enabled without them. This isn't a problem with your new options, which are new. So you can drop all your changes to `SanityCheck.h` here.